src/corefx/System.Globalization.Native/localeStringData.cpp

```diff
@@ -232,11 +253,87 @@ extern "C" int32_t GetLocaleInfoString(const UChar* localeName, LocaleStringData
 			break;
 		default:
 			status = U_UNSUPPORTED_ERROR;
-			assert(false);
 			break;
 	};
 
-	assert(status != U_BUFFER_OVERFLOW_ERROR);
+	return UErrorCodeToBool(status);
+}
+
+/*
+Function:
+NormalizeTimePattern
+
+Convert an ICU non-localized time pattern to .NET format
+*/
+void NormalizeTimePattern(const UnicodeString *srcPattern, UnicodeString *destPattern)
+{
+	// An srcPattern example: "h:mm:ss a"
+	// A destPattern example: "h:mm:ss tt"
+	destPattern->remove();
+
+	for (int i = 0; i <= srcPattern->length() - 1; i++)
+	{
+		UChar ch = srcPattern->charAt(i);
+		switch (ch)
+		{
+		case ':':
+		case '.':
+		case 'H':
+		case 'h':
+		case 'm':
+		case 's':
+			destPattern->append(ch);
+			break;
+
+		case UCHAR_SPACE:
+		case UCHAR_NBSPACE:
+			destPattern->append(UCHAR_SPACE);
+			break;
+
+		case 'a':
```

Comment:
I'll add a check to ignore >1 'a's. Also I did enumerate all time formats on ICU 55 and only found single 'a's